crates/rome_js_formatter/src/builders.rs

```diff
@@ -402,3 +381,95 @@ enum DelimitedMode {
+/// trivias and token, separately.
+///
+/// This function assumes that you will use the token to replicate [format_delimited], which means
+/// that it will add possible line breaks
+pub(crate) fn format_open_delimiter(open_token: &JsSyntaxToken) -> OpenDelimiter {
+    OpenDelimiter::new(open_token)
+}
+
+/// Use this function to create an close delimiter, where you can extract the formatting of
+/// trivias and token, separately.
+///
+/// This function assumes that you will use the token to replicate [format_delimited], which means
+/// that it will add possible line breaks
+pub(crate) fn format_close_delimiter(close_token: &JsSyntaxToken) -> CloseDelimiter {
+    CloseDelimiter::new(close_token)
+}
+
+pub(crate) struct OpenDelimiter<'t> {
+    open_token: &'t JsSyntaxToken,
+}
+
+impl<'t> OpenDelimiter<'t> {
+    pub(crate) fn new(open_token: &'t JsSyntaxToken) -> Self {
+        Self { open_token }
+    }
+
+    /// It extracts the formatted leading trivia of the token, without writing it in the buffer
+    pub(crate) fn format_leading_trivia(&self) -> impl Format<JsFormatContext> + 't {
+        format_leading_trivia(self.open_token)
+    }
+
+    /// It extracts the formatted trailing trivia of the token, without writing it in the buffer
+    pub(crate) fn format_trailing_trivia(&self) -> impl Format<JsFormatContext> + 't {
+        format_with(|f| {
+            // Not really interested in the pre-amble, but want to know if it was written
+            let mut buffer = VecBuffer::new(f.state_mut());
+
+            write!(buffer, [format_trailing_trivia(self.open_token)])?;
+
+            let trivia = buffer.into_vec();
+
+            if !trivia.is_empty() {
+                f.write_elements(trivia)?;
+                soft_line_break().fmt(f)?;
+            }
+
+            Ok(())
+        })
+    }
+
+    /// It extracts the formatted token, without writing it in the buffer
+    pub(crate) fn format_token(&self) -> impl Format<JsFormatContext> + 't {
+        format_with(|f| {
+            f.state_mut().track_token(self.open_token);
+            write!(f, [format_trimmed_token(self.open_token)])
+        })
+    }
+}
+
+pub(crate) struct CloseDelimiter<'t> {
+    close_token: &'t JsSyntaxToken,
+}
+
+impl<'t> CloseDelimiter<'t> {
+    pub(crate) fn new(close_token: &'t JsSyntaxToken) -> Self {
+        Self { close_token }
+    }
+
+    /// It extracts the formatted leading trivia of the token, without writing it in the buffer
+    pub(crate) fn format_trailing_trivia(&self) -> impl Format<JsFormatContext> + 't {
+        format_trailing_trivia(self.close_token)
+    }
+
+    /// It extracts the formatted trailing trivia of the token, without writing it in the buffer
+    pub(crate) fn format_leading_trivia(&self) -> impl Format<JsFormatContext> + 't {
+        format_with(|f| {
+            let mut buffer = PreambleBuffer::new(f, soft_line_break_or_space());
```

Comment:
A `soft_line_break` is sufficient here. 
```suggestion
            let mut buffer = PreambleBuffer::new(f, soft_line_break());
```